pkg/kubecost/asset.go

```diff
@@ -3053,17 +3113,22 @@ func (asr *AssetSetRange) Window() Window {
 func (asr *AssetSetRange) Start() (time.Time, error) {
 	start := time.Time{}
 	firstStartNotSet := true
-	asr.Each(func(i int, as *AssetSet) {
-		as.Each(func(s string, a Asset) {
+
+	asr.RLock()
+	for _, as := range asr.assets {
+		as.RLock()
+		for _, a := range as.assets {
 			if firstStartNotSet {
 				start = a.Start()
 				firstStartNotSet = false
 			}
 			if a.Start().Before(start) {
 				start = a.Start()
 			}
-		})
-	})
+		}
+		as.RUnlock()
+	}
+	asr.RUnlock()
```

Comment:
Avoid `Each` if there exists access to the assets, since it now copies the collection. Additionally, we want to acquire read locks for the internal assets collection. While this change does add read locking for each Asset as well, I'm not sure it's required. Something to look into.